### ISSUE TITLE
CrossEntropy loss with GPU

### ISSUE THREAD

**Issue by alekseidolgikh** [catboost/catboost#1654]
Problem:
I use CrossEntropy loss for training a CatBoost classifier. It works on CPU. But in case of training on GPU I see: "Attempt to call single feature writer on packed feature writer".

catboost version: 0.24.4
Operating System: Windows 10
CPU:
GPU: RTX 3080

**Comment by LyzhinIvan** [comment]
Hi, @alekseidolgikh!
Sorry for long response. Do you remember if you dataset contained text or embedding features?

**Comment by alekseidolgikh** [comment]
Hey, @LyzhinIvan!
Yes, either text (90% of chance) or embedding (10% of chance) features were used.

**Comment by ashimko** [comment]
Hello, I've got the same error with text embeddings in vertion 1.0.3.

**Comment by LyzhinIvan** [comment]
> Hello, I've got the same error with text embeddings in vertion 1.0.3.

Hi, @ashimko!

Could you provide a script with data to reproduce the error?

**Comment by dbulgakov7** [comment]
I get this error when some target feature values are present in the test dataset, but are missing in the training dataset. If you remove these outliers of the target variable from the dataset, the error disappears.

**Comment by ashimko** [comment]
> I get this error when some target values are present in the test dataset but are missing from the training dataset. Or they are in both datasets, but these values are very small - then GPU training does not work. If you remove these target variable outliers from the dataset, the error will disappear.

Yep, had the same issue. It solves this problem.
